homeassistant/components/wled/helpers.py

```diff
@@ -35,3 +35,13 @@ async def handler(self: _WLEDEntityT, *args: _P.args, **kwargs: _P.kwargs) -> No
             raise HomeAssistantError("Invalid response from WLED API") from error
 
     return handler
+
+
+def kelvinTo255(k: int, minK: int, maxK: int) -> int:
```

Comment:
https://github.com/home-assistant/core/blob/e1e64be3c9a2597ca17f77ba1ae6c8dbb063eef3/homeassistant/util/color.py#L535
https://github.com/home-assistant/core/blob/e1e64be3c9a2597ca17f77ba1ae6c8dbb063eef3/homeassistant/util/color.py#L551

It looks like we already have helpers for this

Reply:
hi @bdraco, thank you for the review. I actually found those functions before implementing my own, the problem is that those function return/expect two values (one for cold white, one for warm white); moreover, they convert the K unit to mired unit before performing the actual conversion so there may be an unnecessary performance penalty here. This is why I preferred to implement some simple, specific, functions for this use case. If you still prefer to use those one I will update the code, no problem, please let me know.

Reply:
I don't think the performance issue will be a concern given these functions will not be called frequently.

Reply:
Understood, I'll change the code to use the existing helper, I'll request a new review once done, thank you!

Reply:
Also `_white_levels_to_color_temperature` could be updated to avoid the mireds conversion as its not needed there and left over from when HA used mireds as the default

Reply:
@bdraco I just changed the code (not yet pushed) to use those functions but tests failed. I got slightly different results (e.g. cct of 53 is mapped to 2337 K instead of 2942 K. I suspect this is because the `_white_levels_to_color_temperature ` function expects two different values for the cold white and warm white leds "intensities", while WLED expects a single cct parameter that shifts between warmest and coldest. Do you have any gesttion?

Reply:
In that case it looks like you can't use the helper and will have to use the ones you built.  Bummer that we have to maintain both but it does look like a different method.

Reply:
I'm actually thinking that those functions are indeed bugged. You cannot transform kelvins in mirads, interpolate and then transform back to kelvin, you get a different result. I'm going to stick with my private helper for WLED, when WLED is going to support physical leds color temperature we will get rid of them by using kelvin directly